kratos/processes/tetrahedral_mesh_orientation_check.cpp

```diff
@@ -75,26 +75,31 @@ void TetrahedralMeshOrientationCheck::Execute()
     hashmap faces_map;
 
     for (auto it_cond = mrModelPart.ConditionsBegin(); it_cond != mrModelPart.ConditionsEnd(); it_cond++) {
-        it_cond->Set(VISITED, false); //mark
+        it_cond->Set(VISITED, false); // Mark
 
-        GeometryType& geom = it_cond->GetGeometry();
-        DenseVector<int> ids(geom.size());
+        GeometryType& r_geometry = it_cond->GetGeometry();
 
-        for(IndexType i=0; i<ids.size(); i++) {
-            geom[i].Set(BOUNDARY,true);
-            ids[i] = geom[i].Id();
-        }
+        const GeometryData::KratosGeometryType geometry_type = r_geometry.GetGeometryType();
+
+        if (geometry_type == GeometryData::Kratos_Triangle3D3  || geometry_type == GeometryData::Kratos_Line2D2) {
```

Comment:
This is the change (you check elements, but not conditions)

Reply:
OK. going through the rest of the file I realized that we're checking that the elements are simplex but we're not doing so to ensure that the condition types are the corresponding ones.